src/cargo/core/compiler/build_runner/compilation_files.rs

```diff
@@ -718,8 +718,8 @@ fn compute_metadata(
         }
     }
 
-    let c_metadata = UnitHash(c_metadata_hasher.finish());
-    let c_extra_filename = UnitHash(c_extra_filename_hasher.finish());
+    let c_metadata = UnitHash(Hasher::finish(&c_metadata_hasher));
+    let c_extra_filename = UnitHash(Hasher::finish(&c_extra_filename_hasher));
```

Comment:
Why do we need to make this change in changing the hasher?

Both implement the trait and its in scope.

Reply:
Because it resolves to `StableHasher::finish` first.

https://github.com/rust-lang/rustc-stable-hash/blob/a7bb3b77597092f1743362b6327d4324544d6d95/src/stable_hasher.rs#L167-L169

Do you have other workarounds to make it resolves to `Hasher::finish`?

Reply:
... if their `finish` isn't compatible with regular usage of `Hasher::finish` then thats an odd choice.

How come we can't rely on `FromStableHash`?  ... they have no impls.  *sigh*